examples/lang2/erc721/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+        owned_tokens_count: storage::HashMap<AccountId, u128>,
+        /// Mapping from owner to operator approvals
+        operator_approvals: storage::HashMap<AccountId, storage::HashMap<AccountId, bool>>,
+        /// None Account to validate account
+        none_account: storage::Value<AccountId>,
+    }
+
+    /// Event emitted when a token transfer occurs
+    #[ink(event)]
+    struct Transfer {
+        #[ink(topic)]
+        from: Option<AccountId>,
+        #[ink(topic)]
+        to: AccountId,
+        #[ink(topic)]
+        id: Hash,
+    }
+
+    /// Event emited when a token approve occurs
+    #[ink(event)]
+    struct Approval {
+        #[ink(topic)]
+        from: AccountId,
+        #[ink(topic)]
+        to: AccountId,
+        #[ink(topic)]
+        id: Hash,
+    }
+
+    /// Event emitted when an operator is enabled or disabled for an owner.
+    /// The operator can manage all NFTs of the owner.
+    #[ink(event)]
+    struct ApprovalForAll {
+        #[ink(topic)]
+        owner: AccountId,
+        #[ink(topic)]
+        operator: AccountId,
+        approved: bool
+    }
+
+    impl Erc721 {
+        #[ink(constructor)]
+        fn new(&mut self) {
+            self.none_account.set(AccountId::from([0x0; 32]));
+        }
+
+        /// ===========================
+        /// NOT REQUIRED, JUST FOR TEST
+        /// ===========================
+        #[ink(message)]
+        fn mint(&mut self, id: Hash) -> bool {
+            let caller = self.env().caller();
+            let owner = self.owner_of_or_none(id);
+            if owner != *self.none_account {
+                self.env().println(&format!("Erc721::mint::token is exist"));
+                return false;
+            }
+            self.token_owner.insert(id, caller);
+            let balance = self.balance_of_or_zero(caller);
+            self.owned_tokens_count.insert(caller, balance + 1);
+            self.env().emit_event(Transfer {
+                from: None,
+                to: caller,
+                id: id,
+            });
+            true
+        }
+
+        /// Get token balance of specific account.
+        #[ink(message)]
+        fn balance_of(&self, owner: AccountId) -> Balance {
+            let balance = self.balance_of_or_zero(owner);
+            self.env().println(&format!("Erc721::balance_of(owner = {:?}) = {:?}", owner, balance));
+            balance
+        }
+
+        /// Get owner for specific token.
+        #[ink(message)]
+        fn owner_of(&self, id: Hash) -> AccountId {
+            let owner = self.owner_of_or_none(id);
+            self.env().println(&format!("Erc721::owner_of(token = {:?}) = {:?}", id, owner));
+            owner
+        }
+
+        /// The approved address for this token, or the none address if there is none
+        #[ink(message)]
+        fn get_approved(&self, id: Hash) -> AccountId {
+            let account = self.approved_of_or_none(id);
+            account
```

Comment:
Can be simplified to just `self.approved_of_or_none(id)`

Reply:
Also instead of none account you could use `Option<AccountId>` as return type which would be more idiomatic Rust.